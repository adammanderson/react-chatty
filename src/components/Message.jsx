import React from 'react'
import PropTypes from 'prop-types'
import styled from 'react-emotion'
import Bubble from './Bubble'
import Avatar from './Avatar'

const MessageWrapper = styled.div`
  display: flex;
  flex-flow: ${props => props.type === 'agent' ? 'row nowrap' : 'row-reverse nowrap'};
  margin-bottom: 1.8em;
  align-items: flex-end;
`

const Message = (props) => (
  <MessageWrapper type={props.message.type}>
    <Avatar />
    <Bubble message={props.message} />
  </MessageWrapper>
)

Message.propTypes = {
  message: PropTypes.shape({
    type: PropTypes.oneOf(['user', 'agent', 'waiting']).isRequired,
    content: PropTypes.string.isRequired
  }).isRequired
}

export default Message
