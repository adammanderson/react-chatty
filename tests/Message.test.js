import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'emotion-theming'
import THEMES from '../src/themes'
import Message from '../src/components/Message'

const testMessage =  {
    type: 'user',
    content: '**Here\'s the forecast: Tonight will start dry with clear periods.** Cloud may thicken through the last few hours of the night. A gentle southwesterly breeze can be expected.',
    time: '2018-09-13T14:10:13'
}

test('Message renders with theme', () => {
  const component = renderer.create(
    <ThemeProvider theme={THEMES.light}>
      <Message
        message={testMessage}
      />
    </ThemeProvider>
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
