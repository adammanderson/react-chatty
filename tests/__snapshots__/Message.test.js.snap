// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bubble renders with theme 1`] = `
<div
  className="css-1pt9r9f e12o4ec70"
  type="user"
>
  <div
    className="css-h1bspy ebzut8o0"
  />
  <div
    className="css-hth5us e193rrgr0"
    type="user"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p><strong>Here&#39;s the forecast: Tonight will start dry with clear periods.</strong> Cloud may thicken through the last few hours of the night. A gentle southwesterly breeze can be expected.</p>
",
        }
      }
    />
    <div
      className="css-1fwbb47 e193rrgr1"
    >
      <time
        dateTime={"2018-09-13T13:10:13.000Z"}
      >
        14:10pm
      </time>
    </div>
  </div>
</div>
`;
